File: android/src/main/java/com/evotor/egais/mapper/ActChargeOnShop.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.evotor.egais.converter.QuantityBigDecimalConverter
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

internal fun createActChargeOnShop(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("docOwner", cursor.getString(cursor.getColumnIndexOrThrow("OWNER")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("number", cursor.getString(cursor.getColumnIndexOrThrow("NUMBER")))
    result.putString("actDate", cursor.getString(cursor.getColumnIndexOrThrow("ACT_DATE")))
    result.putString("type", cursor.getString(cursor.getColumnIndexOrThrow("TYPE_ACT_CHARGE_ON_SHOP")))
    result.putString("actWriteOff", cursor.getString(cursor.getColumnIndexOrThrow("ACT_WRITE_OFF")))
    result.putString("note", cursor.getString(cursor.getColumnIndexOrThrow("NOTE")))
    result.putString("status", cursor.getString(cursor.getColumnIndexOrThrow("STATUS")))
    result.putString("rejectComment", cursor.getString(cursor.getColumnIndexOrThrow("REJECT_COMMENT")))
    result.putString("replyId", cursor.getString(cursor.getColumnIndexOrThrow("REPLY_ID")))
    return result
}

internal fun createActChargeOnShopPosition(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("actChargeOnShopUuid", cursor.getString(cursor.getColumnIndexOrThrow("ACT_CHARGE_ON_SHOP_UUID")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putDouble("quantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("QUANTITY"))).toDouble())
    result.putMap("productInfo", createProductInfo(cursor))
    return result
}

File: model/document/ActChargeOnShop.js
/**
 * Классы и типы для работы с актами постановки на баланс.
 * @module module:actChargeOn
 */
import ProductInfo from "../dictionary/ProductInfo"
import {ActChargeOnShopStatus, ActChargeOnShopType} from '../types'

/**
 * @class module:actChargeOn.ActChargeOnShop
 * @classdesc Акт постановки на баланс в торговый зал.
 * @param {string} uuid Уникальный идентификатор
 * @param {string} docOwner Кто подает документы
 * @param {?string} identity ID акта (клиентский)
 * @param {?string} number Номер документа
 * @param {Date} actDate Дата постановски на баланс
 * @param {module:actChargeOn#ActChargeOnShopType} type Причина постановки на баланс
 * @param {?string} actWriteOff Регистрационный номер акта списания по типу "Пересортица". Обязательно для заполнения для типа постановки на баланс "Пересортица".
 * @param {?string} note Примечание
 * @param {module:actChargeOn#ActChargeOnShopStatus} status Статус документа
 * @param {?string} rejectComment Комментарий при отказе
 * @param {?string} replyId Уникальный идентификатор документа (присваивается УТМ); совпадает с идентификатором исходящего документа, который получили в ответе
 */
export default class ActChargeOnShop {
    constructor(uuid: string,
                docOwner: string,
                identity: (string | null),
                number: string,
                actDate: Date,
                type: ActChargeOnShopType,
                actWriteOff: (string | null),
                note: (string | null),
                status: ActChargeOnShopStatus,
                rejectComment: (string | null),
                replyId: (string | null)) {
        this.uuid = uuid;
        this.docOwner = docOwner;
        this.identity = identity;
        this.number = number;
        this.actDate = actDate;
        this.type = type;
        this.actWriteOff = actWriteOff;
        this.note = note;
        this.status = status;
        this.rejectComment = rejectComment;
        this.replyId = replyId;
    }
}

/**
 * @class module:actChargeOn.ActChargeOnShopPosition
 * @classdesc Позиция в акте постановки на баланс в торговый зал.
 * @param {string} uuid Уникальный идентификатор позиции акта
 * @param {string} actChargeOnShopUuid Идентификатор акта постановки, содержащего позицию
 * @param {string} identity Идентификатор позиции внутри накладной
 * @param {number} quantity Количество
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукции
 */
export class ActChargeOnShopPosition {
    constructor(uuid: string,
                actChargeOnShopUuid: string,
                identity: string,
                quantity: number,
                productInfo: ProductInfo) {
        this.uuid = uuid;
        this.actChargeOnShopUuid = actChargeOnShopUuid;
        this.identity = identity;
        this.quantity = quantity;
        this.productInfo = productInfo;
    }
}